RTools 4.0 Windows issues warnings on build

## Description
As the title says, cmdstan on Windows with RTools 4.0 issues a bunch of warnings. 

It all still works, but if someone has ideas on whether we can clean these up that would be great.

On cmdstan build we get
```
In file included from stan/lib/stan_math/stan/math/rev/core.hpp:55,
                 from stan/src/stan/model/model_base.hpp:5,
                 from src/cmdstan/command.hpp:23,
                 from src/cmdstan/main.cpp:2:
stan/lib/stan_math/stan/math/rev/core/std_isinf.hpp:16:43: warning: optimization attribute on 'bool std::isinf(const stan::math::var&)' follows definition but the attribute doesn't match [-Wattributes]
 inline bool isinf(const stan::math::var& a) {
                                           ^
In file included from stan/lib/stan_math/lib/boost_1.72.0/boost/config/no_tr1/cmath.hpp:21,
                 from stan/lib/stan_math/lib/boost_1.72.0/boost/numeric/conversion/converter_policies.hpp:16,
                 from stan/lib/stan_math/lib/boost_1.72.0/boost/numeric/conversion/converter.hpp:14,
                 from stan/lib/stan_math/lib/boost_1.72.0/boost/numeric/conversion/cast.hpp:33,
                 from stan/lib/stan_math/lib/boost_1.72.0/boost/lexical_cast/detail/converter_numeric.hpp:36,
                 from stan/lib/stan_math/lib/boost_1.72.0/boost/lexical_cast/try_lexical_convert.hpp:43,
                 from stan/lib/stan_math/lib/boost_1.72.0/boost/lexical_cast.hpp:32,
                 from stan/src/stan/callbacks/writer.hpp:4,
                 from src/cmdstan/arguments/argument.hpp:4,
                 from src/cmdstan/arguments/argument_parser.hpp:4,
                 from src/cmdstan/command.hpp:4,
                 from src/cmdstan/main.cpp:2:
C:/rtools40/mingw64/include/c++/8.3.0/cmath:584:3: note: previous definition of 'constexpr bool std::isinf(float)' was here
   isinf(float __x)
   ^~~~~

In file included from stan/lib/stan_math/lib/tbb_2019_U8/include/tbb/task_arena.h:20,
                 from stan/lib/stan_math/stan/math/prim/functor/reduce_sum.hpp:7,
                 from stan/lib/stan_math/stan/math/prim/functor.hpp:21,
                 from stan/lib/stan_math/stan/math/prim.hpp:11,
                 from stan/src/stan/io/dump.hpp:6,
                 from src/cmdstan/command.hpp:20,
                 from src/cmdstan/main.cpp:2:
stan/lib/stan_math/lib/tbb_2019_U8/include/tbb/task.h:1065:84: warning: optimization attribute on 'void* operator new(size_t, const tbb::internal::allocate_root_proxy&)' follows definition but the attribute doesn't match [-Wattributes]
 inline void *operator new( size_t bytes, const tbb::internal::allocate_root_proxy& ) {
                                                                                    ^
In file included from C:/rtools40/mingw64/include/c++/8.3.0/bits/exception_ptr.h:40,
                 from C:/rtools40/mingw64/include/c++/8.3.0/exception:143,
                 from C:/rtools40/mingw64/include/c++/8.3.0/ios:39,
                 from C:/rtools40/mingw64/include/c++/8.3.0/ostream:38,
                 from C:/rtools40/mingw64/include/c++/8.3.0/iterator:64,
                 from stan/lib/stan_math/lib/boost_1.72.0/boost/iterator/iterator_traits.hpp:10,
                 from stan/lib/stan_math/lib/boost_1.72.0/boost/range/iterator_range_core.hpp:26,
                 from stan/lib/stan_math/lib/boost_1.72.0/boost/lexical_cast.hpp:30,
                 from stan/src/stan/callbacks/writer.hpp:4,
                 from src/cmdstan/arguments/argument.hpp:4,
                 from src/cmdstan/arguments/argument_parser.hpp:4,
                 from src/cmdstan/command.hpp:4,
                 from src/cmdstan/main.cpp:2:
C:/rtools40/mingw64/include/c++/8.3.0/new:168:14: note: previous definition of 'void* operator new(std::size_t, void*)' was here
 inline void* operator new(std::size_t, void* __p) _GLIBCXX_USE_NOEXCEPT
              ^~~~~~~~
			  
In file included from stan/lib/stan_math/lib/tbb_2019_U8/include/tbb/task_arena.h:20,
                 from stan/lib/stan_math/stan/math/prim/functor/reduce_sum.hpp:7,
                 from stan/lib/stan_math/stan/math/prim/functor.hpp:21,
                 from stan/lib/stan_math/stan/math/prim.hpp:11,
                 from stan/src/stan/io/dump.hpp:6,
                 from src/cmdstan/command.hpp:20,
                 from src/cmdstan/main.cpp:2:
stan/lib/stan_math/lib/tbb_2019_U8/include/tbb/task.h:1103:101: warning: optimization attribute on 'void operator delete(void*, const tbb::internal::allocate_additional_child_of_proxy&)' follows definition but the attribute doesn't match [-Wattributes]
 inline void operator delete( void* task, const tbb::internal::allocate_additional_child_of_proxy& p ) {
                                                                                                     ^
In file included from C:/rtools40/mingw64/include/c++/8.3.0/bits/exception_ptr.h:40,
                 from C:/rtools40/mingw64/include/c++/8.3.0/exception:143,
                 from C:/rtools40/mingw64/include/c++/8.3.0/ios:39,
                 from C:/rtools40/mingw64/include/c++/8.3.0/ostream:38,
                 from C:/rtools40/mingw64/include/c++/8.3.0/iterator:64,
                 from stan/lib/stan_math/lib/boost_1.72.0/boost/iterator/iterator_traits.hpp:10,
                 from stan/lib/stan_math/lib/boost_1.72.0/boost/range/iterator_range_core.hpp:26,
                 from stan/lib/stan_math/lib/boost_1.72.0/boost/lexical_cast.hpp:30,
                 from stan/src/stan/callbacks/writer.hpp:4,
                 from src/cmdstan/arguments/argument.hpp:4,
                 from src/cmdstan/arguments/argument_parser.hpp:4,
                 from src/cmdstan/command.hpp:4,
                 from src/cmdstan/main.cpp:2:
C:/rtools40/mingw64/include/c++/8.3.0/new:174:13: note: previous definition of 'void operator delete(void*, void*)' was here
 inline void operator delete  (void*, void*) _GLIBCXX_USE_NOEXCEPT { }
             ^~~~~~~~
```

There are also some that come up when compiling a model:
```
In file included from stan/lib/stan_math/lib/tbb_2019_U8/include/tbb/task_arena.h:20,
                 from stan/lib/stan_math/lib/tbb_2019_U8/include/tbb/task_scheduler_observer.h:22,
                 from stan/lib/stan_math/stan/math/rev/core/init_chainablestack.hpp:6,
                 from stan/lib/stan_math/stan/math/rev/core.hpp:10,
                 from stan/lib/stan_math/stan/math/rev.hpp:6,
                 from stan/lib/stan_math/stan/math.hpp:19,
                 from stan/src/stan/model/model_header.hpp:4,
                 from examples/bernoulli/bernoulli.hpp:3:
stan/lib/stan_math/lib/tbb_2019_U8/include/tbb/task.h:1103:101: warning: optimization attribute on 'void operator delete(void*, const tbb::internal::allocate_additional_child_of_proxy&)' follows definition but the attribute doesn't match [-Wattributes]
 inline void operator delete( void* task, const tbb::internal::allocate_additional_child_of_proxy& p ) {
                                                                                                     ^
In file included from stan/lib/stan_math/lib/eigen_3.3.7/Eigen/Core:82,
                 from stan/lib/stan_math/lib/eigen_3.3.7/Eigen/Dense:1,
                 from stan/lib/stan_math/stan/math/prim/fun/Eigen.hpp:22,
                 from stan/lib/stan_math/stan/math/rev.hpp:4,
                 from stan/lib/stan_math/stan/math.hpp:19,
                 from stan/src/stan/model/model_header.hpp:4,
                 from examples/bernoulli/bernoulli.hpp:3:
C:/rtools40/mingw64/include/c++/8.3.0/new:174:13: note: previous definition of 'void operator delete(void*, void*)' was here
 inline void operator delete  (void*, void*) _GLIBCXX_USE_NOEXCEPT { }
             ^~~~~~~~

```
#### Current Version:
v3.2.0

--- a/make/compiler_flags
+++ b/make/compiler_flags
@@ -161,6 +161,7 @@ endif
 
 ## makes reentrant version lgamma_r available from cmath
 CXXFLAGS_OS += -D_REENTRANT
+CXXFLAGS_WARNINGS += -Wno-int-in-bool-context -Wno-attributes
 
 ################################################################################
 # Setup OpenCL

--- a/make/compiler_flags
+++ b/make/compiler_flags
@@ -161,7 +161,10 @@ endif
 
 ## makes reentrant version lgamma_r available from cmath
 CXXFLAGS_OS += -D_REENTRANT
+
+ifeq (gcc,$(CXX_TYPE))
 CXXFLAGS_WARNINGS += -Wno-int-in-bool-context -Wno-attributes
+endif
 
 ################################################################################
 # Setup OpenCL

